admin_ui/src/components/CSVButton.vue

```diff
@@ -0,0 +1,94 @@
+<template>
+    <a class="button">
+        <downloadexcel
+            :before-finish="finishDownload"
+            :before-generate="startDownload"
+            :fetch="fetchExportedRows"
+            :name="tableName + '.csv'"
+            type="csv"
+        >
+            <span>
+                <font-awesome-icon icon="file-csv" />Export as CSV
+            </span>
+        </downloadexcel>
+    </a>
+</template>
+
+<script lang="ts">
+import axios from "axios"
+import downloadexcel from "vue-json-excel"
+import * as i from "../interfaces"
+
+export default {
+    props: ["tableName"],
+    data() {
+        return {
+            tableName: this.tableName,
+        }
+    },
+    components: {
+        downloadexcel,
+    },
+    methods: {
+        // Export data as csv from json:
+        async fetchExportedRows() {
+            const params = this.$store.state.filterParams
+            const tableName = this.$store.state.currentTableName
+            const sortBy = this.$store.state.sortBy
+            if (sortBy) {
+                let prefix = sortBy.ascending ? "" : "-"
+                params["__order"] = prefix + sortBy.property
+            }
+            // Get the row counts:
+            const response = await axios.get(`api/tables/${tableName}/count/`, {
+                params,
+            })
+            const data = response.data as i.RowCountAPIResponse
+            const localParams = { ...params }
+
+            localParams["__page"] = data.count
+            // Set higher __page_size param to have fewer requests to api:
+            localParams["__page_size"] = 1000
+            const pages = Math.ceil(data.count / localParams["__page_size"])
+            const exportedRows = []
+
+            try {
+                for (let i = 1; i < pages + 1; i++) {
+                    localParams["__page"] = i
+                    const response = await axios.get(
+                        `api/tables/${tableName}/?__readable=true`,
+                        {
+                            params: localParams,
+                        }
+                    )
+                    exportedRows.push(...response.data.rows)
+                }
+                return exportedRows
+            } catch (error) {
+                console.log(error.response)
+            }
+        },
+        async startDownload() {
+            const params = this.$store.state.filterParams
+            const tableName = this.$store.state.currentTableName
+            const response = await axios.get(`api/tables/${tableName}/count/`, {
+                params,
+            })
+            const data = response.data as i.RowCountAPIResponse
+            if (data.count > 10000) {
+                alert(
+                    "Number of rows is really high. Do you want to start download your data."
+                )
```

Comment:
We need to change this to:

```javascript
        if confirm(
                "The number of rows is really high. Do you want to start downloading your data?"
         ) {
             this.startDownload
         }
```
So when they click OK, it starts downloading.

Reply:
But data is start download when click export button and `confirm()` doesn't change anything. I think that we should go simpler

```
startDownload() {
    alert("Your data is start downloading. For large dataset download may take few seconds.")
},
finishDownload() {
	alert("Your data is ready.")
},
```

With that we nicely inform user that large dataset need few seconds to download and we save one API call because don't need rows count in `startDownload()`.  What do you think about that?

Reply:
Ah, I see - didn't realise that.